web/modules/custom/mymodule/src/Plugin/Block/Temperatureblock.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Drupal\mymodule\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+use GuzzleHttp\Client;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a 'Temperature' Block.
+ *
+ * @Block(
+ *   id = "temperature_block",
+ *   admin_label = @Translation("Temperature block"),
+ * )
+ */
+class Temperatureblock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * Object variable.
+   *
+   * @var object
+   */
+  protected $configfactory;
+
+  /**
+   * Construct method.
+   *
+   * @param array $configuration
+   *   It return value of array.
+   * @param string $plugin_id
+   *   It return plugin id.
+   * @param mixed $plugin_definition
+   *   It returns the plugin definition.
+   * @param Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   It returns the object.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ConfigFactoryInterface $config_factory) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+
+    $this->configfactory = $config_factory;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('config.factory'),
+    );
+  }
+
+  /**
+   * Build function to render outut of the block.
+   *
+   * @return array
+   *   It will return the array.
+   */
+  public function build() {
+    $config = $this->configfactory->get('mymodule.settings');
+
+    $country = $config->get('country');
+    $city = $config->get('city');
+    $apikey = $config->get('apikey');
+    $apiendpoint = $config->get('apiendpoint');
+    $client = new Client();
```

Comment:
@anitha-innoraft - Please use `http_client` service for external requests.

Reply:
Done